src/key_io.cpp

```diff
@@ -341,3 +341,21 @@ std::vector<unsigned char> DecodeSilentAddress(const std::string& str)
     if ((version == 0 && silent_payment_data.size() != SILENT_PAYMENT_V0_DATA_SIZE) || silent_payment_data.size() < SILENT_PAYMENT_V0_DATA_SIZE) return {};
     return silent_payment_data;
 }
+
+std::string EncodeSilentDestination(const CPubKey& scan_pubkey, const CPubKey& spend_pubkey)
+{
```

Comment:
328951d056e939b6db573043fc71d8bd6ae68dff: if cheap, could assert/assume both ` CPubKey` are valid.

Reply:
This now uses the `src/key_io.cpp::Encode` function, which takes a `V0SilentPaymentDestination` object, so should be good. If we do want an assert, we can/should do it when creating the `V0SilentPaymentDestination` object